common/messages.py

```diff
@@ -361,8 +362,9 @@ def __init__(self, game_setting_mode: str):
 class Login2ClientMenuData(Message):
     msg_id = _MSGID_LOGIN2CLIENT_MENUDATA
 
-    def __init__(self, menu_item):
+    def __init__(self, menu_item, timestamp_value: datetime.datetime):
         self.menu_item = menu_item
+        self.timestamp = timestamp_value.isoformat()
```

Comment:
This representation depends on the timezone of the login server. Do you only use it for ordering messages?

Reply:
Although the timestamp does ultimately come from `utcnow()`, it isn't even used for ordering. The client uses it to denote the 'batch' of updates. i.e. if it receives a message from with a different timestamp it clears everything it has and starts again.

Basically it's to mean re-sending the data would replace rather than add to what was already there.